src/rust/engine/engine_cffi/src/lib.rs

```diff
@@ -390,9 +389,8 @@ fn make_core(
   )
 }
 
-fn workunits_to_py_tuple_value(workunits: &HashSet<WorkUnit>) -> Value {
+fn workunits_to_py_tuple_value<'a>(workunits: &mut impl Iterator<Item = &'a WorkUnit>) -> Value {
```

Comment:
```suggestion
fn workunits_to_py_tuple_value<'a>(workunits: impl Iterator<Item = &'a WorkUnit>) -> Value {
```

seems strange to take a `&mut` to this thing which we're going to fully consume?